packages/extension-api/src/extension-api.d.ts

```diff
@@ -136,6 +136,37 @@ declare module '@podman-desktop/api' {
     readonly storagePath: string;
   }
 
+  /**
+   * A provider result represents the values a provider, like the {@linkcode ImageCheckerProvider},
+   * may return. For once this is the actual result type `T`, like `ImageChecks`, or a Promise that resolves
+   * to that type `T`. In addition, `null` and `undefined` can be returned - either directly or from a
+   * Promise.
+   *
+   * The snippets below are all valid implementations of the {@linkcode ImageCheckerProvider}:
+   *
+   * ```ts
+   * let a: ImageCheckerProvider = {
+   *  check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
+   *    return new ImageChecks();
+   *  }
+   *
+   * let b: ImageCheckerProvider = {
+   *  check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
+   * 		return new Promise(resolve => {
+   * 			resolve(new ImageChecks());
+   * 	 	});
+   * 	}
+   * }
+   *
+   * let c: ImageCheckerProvider = {
+   *  check(image: ImageInfo, token?: CancellationToken): ProviderResult<ImageChecks> {
+   * 		return; // undefined
+   * 	}
+   * }
+   * ```
+   */
+  export type ProviderResult<T> = T | undefined | null | Promise<T | undefined | null>;
```

Comment:
```suggestion
  export type ProviderResult<T> = T | undefined | Promise<T | undefined>;
```

In Podman Desktop we've banned `null`